### PR TITLE
[docs] Updating Kubernetes upgrade instruction

### PR DESCRIPTION
**Pull request by framsouza**
- Updating Vault Upgrade instructions on Kubernetes to avoid leader election while deleting one pod
- Potential cause/fix for [1.15.4] Raft leader election unexpected behavior  #24995

### CODE REVIEW

File: website/content/docs/platform/k8s/helm/run.mdx
Comment (schavis) [inline website/content/docs/platform/k8s/helm/run.mdx:441]:
```suggestion

If you deployed Vault in high availability (`ha`) mode, you must upgrade your
standby pods before upgrading the active pod. 

Before deleting the standby pod, remove the associated node from the raft with
`vault operator raft remove-peer <server_id>`.

To confirm that the node was successfully removed from Raft, run
`vault operator raft list-peers`.

Once you confirm the removal succeeded, you can delete the pod.

<Warning title="Delete nodes to avoid unnecessary leader elections">

Removing a pod without deleting the node from its cluster first means that the
number of nodes Raft
will be unaware of the correct number of nodes in the cluster. This can trigger a leader election, potentially causing downtime.

</Warning>
```

Style correction: write in active voice, make explicit instructions easy to see at a glance.

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/platform/k8s/helm/run.mdx
Comment (schavis) [inline website/content/docs/platform/k8s/helm/run.mdx:455]:
```suggestion
Removing a pod without first deleting the node from its cluster means that
Raft will be aware of the correct number of nodes in the cluster. Not knowing
the correct number of nodes can trigger a leader election, which can potentially
cause unneeded downtime.
```
Style correction: writein active voice, avoid "this" as a pronoun

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/platform/k8s/helm/run.mdx
Comment (schavis) [inline website/content/docs/platform/k8s/helm/run.mdx:454]:
```suggestion
Raft will not be aware of the correct number of nodes in the cluster. Not knowing
```

I'm guessing that should be "will not"?

File: website/content/docs/platform/k8s/helm/run.mdx
Comment (schavis) [inline website/content/docs/platform/k8s/helm/run.mdx:441]:
```suggestion
standby pods before upgrading the active pod:
```

File: website/content/docs/platform/k8s/helm/run.mdx
Comment (schavis) [inline website/content/docs/platform/k8s/helm/run.mdx:449]:
```suggestion
1. Before deleting the standby pod, remove the associated node from the raft
   with `vault operator raft remove-peer <server_id>`.
1. Confirm Vault removed the node successfully from Raft with
   `vault operator raft list-peers`.
1. Once you confirm the removal, delete the pod.
```

Suggest making it obvious these are instructions by numbering them

**Review by schavis — APPROVED** [review]
Approving so you won't need me to re-review, but please apply the suggestions before merging 🙇🏽‍♀️

**Review by VioletHynes — APPROVED** [review]
Looks good! I applied Sarah's suggestions and I'll try and get this merged!

**Pull request merged**
